### PR TITLE
Add web3 talk

### PR DESCRIPTION
**Pull request by mawalu**

### CODE REVIEW

**Comment by rad4day** [comment]
Files Changed: 63
🤨

ist das wenigstens reusable? :D

File: talks/web3/slides.md [hunk 1]
```diff
@@ -0,0 +1,326 @@
...
+### Tools: web3.* & hardhat
+
+ * Usefull for more advanced exploits
+ * Provide lots of dev tooling
+
+---
+
+### Challenges
+
+ * Intro chals from Slack
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:322]: Could this be improves in some way?
I.e. is the source of the KITCTFCT challenge public or already deployed somewhere?

**Comment by mawalu** [comment]
> Files Changed: 63 raised_eyebrow
> 
> ist das wenigstens reusable? :D

War zumindest die Idee. reveal.js ist extra in nem eigenen Ordner damit man das auch für andere Slides nutzen kann

File: talks/web3/slides.md [hunk 2]
```diff
@@ -0,0 +1,326 @@
...
+     * Contract address is target. Call options as txdata
+     * State can be updated
+ * Invalid contract call
+     * Contract code triggers revert
+
+---
+
+### Solidity
+
+ * Lanuage used in the ethereum ecosystem
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:67]:
```suggestion
 * Language used in the Ethereum ecosystem
```

File: talks/web3/slides.md [hunk 3]
```diff
@@ -0,0 +1,326 @@
+## Smartcontract Exploitation
+
+Martin
+
+---
+
+### Overview
+
+ * Big Picture: Blockchains and Smartcontracts
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:9]:
```suggestion
 * Big Picture: Blockchains and smart contracts
```

File: talks/web3/slides.md [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+
+----
+
+### Blockchain: Visual
+
+![](https://c.m5w.de/uploads/65dd4895-47b7-4b90-91c2-a93afebbb177.png)
+
+----
+
+### Smartcontracts
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:43]:
```suggestion
### Smart Contracts
```

File: talks/web3/slides.md [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+
+### Blockchain
+
+ * Distributed system with shared state
+ * Idea: "Money is Memory"
+ * Distribution of coins as a list of transactions
+     * State can be verified
+     * "Just" need to agree on the transactions
+ * Blocks: chunk of new transactions
+ * Keypairs as accounts
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:24]:
```suggestion
 * Key pairs as accounts
```

File: talks/web3/slides.md [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+
+ * Focus on Ethereum / EVM
+ * Extend the idea: store and modify arbitrary state instead of balances
+ * New type of users: programs
+ * Like transfers: need to be deterministic
+     * Run by every node to compute state update
+
+----
+
+### Smartcontracts: Transactions
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:53]:
```suggestion
### Smart Contracts: Transactions
```

File: talks/web3/slides.md [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+
+### Solidity
+
+ * Lanuage used in the ethereum ecosystem
+     * object-oriented, high-level language for implementing smart contracts
+     * influenced by C++, Python and JavaScript
+     * statically typed, supports inheritance, libraries and complex user-defined types
+ * Compiles to EVM bytecode
+     * Non-ethereum EVM chains exist
+ * Smartcontracts can use other languages (or handcraft EVM bytecode)
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:73]:
```suggestion
 * Smart contracts can use other languages (or handcraft EVM bytecode)
```

File: talks/web3/slides.md [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+    function stuff() public {
+        // lots of code, expensive to store
+        test = 1337; // A.test = 1337
+    }
+}
+```
+
+----
+
+### Solidity: "randomness"
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:217]:
```suggestion
### Solidity: "Randomness"
```

File: talks/web3/slides.md [hunk 9]
```diff
@@ -0,0 +1,326 @@
+## Smartcontract Exploitation
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:1]:
```suggestion
## Smart Contract Exploitation
```

File: talks/web3/slides.md [hunk 10]
```diff
@@ -0,0 +1,326 @@
...
+ * https://solidity-by-example.org/
+ * https://learnxinyminutes.com/docs/solidity/
+ * https://docs.soliditylang.org
+
+---
+
+### Blockchains in CTFs
+
+ * Transactions cost money: using real chains is expensive
+ * Proof-of-Work costs energy: wastefull
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:271]:
```suggestion
 * Proof-of-Work costs energy: wasteful
```

File: talks/web3/slides.md [hunk 11]
```diff
@@ -0,0 +1,326 @@
...
+        return uint256(blockhash(block.number-1));
+    }
+}
+```
+
+----
+
+### Solidity: Interfaces
+
+ * ERC: standards & interfaces for smartcontracts
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:231]:
```suggestion
 * ERC: standards & interfaces for smart contracts
```

File: talks/web3/slides.md [hunk 12]
```diff
@@ -0,0 +1,326 @@
...
+ * Can create multiple accounts / keys
+ * Can export keys
+ * Talks to mainnet node by default, supports custom RPCs
+
+----
+
+### Tools: Remix
+
+ * IDE + Debug environment
+ * Usefull to write "exploit contracts"
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:307]:
```suggestion
 * Useful to write "exploit contracts"
```

File: talks/web3/slides.md [hunk 13]
```diff
@@ -0,0 +1,326 @@
...
+
+ * Usefull for more advanced exploits
+ * Provide lots of dev tooling
+
+---
+
+### Challenges
+
+ * Intro chals from Slack
+ * KITCTFCTF ethereum challenge
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:323]:
```suggestion
 * KITCTFCTF Ethereum challenge
```

File: talks/web3/slides.md [hunk 14]
```diff
@@ -0,0 +1,326 @@
...
+ * IDE + Debug environment
+ * Usefull to write "exploit contracts"
+ * EVM environment in the browser
+ * Can use "injected metamask" provider to talk to custom chains
+
+----
+
+### Tools: web3.* & hardhat
+
+ * Usefull for more advanced exploits
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:315]:
```suggestion
 * Useful for more advanced exploits
```

File: talks/web3/slides.md [hunk 15]
```diff
@@ -0,0 +1,326 @@
...
+}
+
+function printMoney() public onlyOnwer {
+    // ...
+}
+```
+
+----
+
+### Solidity: Calls & transfers
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:140]:
```suggestion
### Solidity: Calls & Transfers
```

File: talks/web3/slides.md [hunk 16]
```diff
@@ -0,0 +1,326 @@
...
+    
+    fallback() external payable {
+        e.cashout(1 ether);
+    }
+}
+```
+
+----
+
+### Solidity: Delegate call
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:192]:
```suggestion
### Solidity: Delegate Call
```

File: talks/web3/slides.md [hunk 17]
```diff
@@ -0,0 +1,326 @@
...
+### Solidity: Interfaces
+
+ * ERC: standards & interfaces for smartcontracts
+     * ERC20 popular interface for tokens
+     * ERC721 interface for NFTs
+ * FOSS implementations exists and can be used
+
+----
+
+### Solidity: Real world token
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:238]:
```suggestion
### Solidity: Real World Token
```

File: talks/web3/slides.md [hunk 18]
```diff
@@ -0,0 +1,326 @@
...
+
+### Blockchains in CTFs
+ * Sometimes public testnets are used
+     * Transactions are public: solves are public
+ * Faucets are provided to get coins for free
+ * Sometimes remote provisions contract instance for every player
+
+----
+
+### Tools to use
```
Comment (intrigus-lgtm) [inline talks/web3/slides.md:286]:
```suggestion
### Tools to Use
```

**Review by intrigus-lgtm — COMMENTED** [review]
Tried to incorporate suggestions from @rad4day where they made sense to me.
Also tried to change all titles to use title case; happy to revert that if you don't like it.

File: talks/web3/index.html [hunk 19]
```diff
@@ -0,0 +1,40 @@
+<!doctype html>
+<html lang="en">
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+
+		<title>KITCTF - Smartcontract Exploitation</title>
```
Comment (sevenmaster) [inline talks/web3/index.html:7]: <title>KITCTF – Smart Contract Exploitation</title>

File: talks/web3/index.html [hunk 20]
```diff
@@ -0,0 +1,40 @@
...
+		<link rel="stylesheet" href="../reveal/dist/reveal.css">
+		<link rel="stylesheet" href="../reveal/dist/theme/black.css">
+
+		<!-- Theme used for syntax highlighted code -->
+		<link rel="stylesheet" href="../reveal/plugin/highlight/monokai.css">
+	</head>
+	<body>
+		<div class="reveal">
+			<div class="slides">
+        <section data-separator="^(\r\n?|\n)---(\r\n?|\n)$" data-separator-vertical="^(\r\n?|\n)----(\r\n?|\n)$" data-markdown="./slides.md">
```
Comment (sevenmaster) [inline talks/web3/index.html:19]: Why is there a question mark, behind the "\n"?
Where is a single carriage return considered a valid line break?

File: talks/web3/index.html (hunk 19 above)
Comment (intrigus-lgtm) [reply talks/web3/index.html:7]:
```suggestion
		<title>KITCTF – Smart Contract Exploitation</title>
```

Comment (intrigus-lgtm) [inline talks/web3/index.html:7]:
```suggestion
		<title>KITCTF – Smart Contract Exploitation</title>
```

File: talks/web3/index.html [hunk 21]
```diff
@@ -0,0 +1,40 @@
...
+<html lang="en">
+	<head>
+		<meta charset="utf-8">
+		<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">
+
+		<title>KITCTF - Smartcontract Exploitation</title>
+
+		<link rel="stylesheet" href="../reveal/dist/reset.css">
+		<link rel="stylesheet" href="../reveal/dist/reveal.css">
+		<link rel="stylesheet" href="../reveal/dist/theme/black.css">
```
Comment (intrigus-lgtm) [inline talks/web3/index.html:11]:
```suggestion
		<link rel="stylesheet" href="../reveal/dist/tdddheme/black.css">
```
This is a test.

File: talks/web3/index.html (hunk 20 above)
Comment (mawalu) [reply talks/web3/index.html:19]: Good question. I just ripped that entire line from the codimd codebase

**Pull request merged**